pkg/operator/controller.go

```diff
@@ -135,6 +139,81 @@ func (c *Controller) createOrUpdateResources(cr *imageregistryv1.Config) error {
 }
 
 func (c *Controller) sync() error {
+	prunerResourceFound := true
+	pcr, err := c.listers.ImagePrunerConfigs.Get(defaults.ImageRegistryImagePrunerResourceName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			prunerResourceFound = false
+		} else {
+			return fmt.Errorf("failed to get %q image registry pruner resource: %s", defaults.ImageRegistryImagePrunerResourceName, err)
+		}
+	}
+	if prunerResourceFound {
+		pcr = pcr.DeepCopy() // we don't want to change the cached version
+		prevPCR := pcr.DeepCopy()
+		prunerCronJob, err := c.listers.CronJobs.Get("image-pruner")
+		if errors.IsNotFound(err) {
+			prunerCronJob = nil
+		} else if err != nil {
+			return fmt.Errorf("failed to get image-pruner pruner job: %s", err)
+		} else {
+			prunerCronJob = prunerCronJob.DeepCopy()
+		}
+
+		prunerJobs, err := c.listers.Jobs.List(labels.Everything())
```

Comment:
List _all_ Jobs in the namespace? In general I don't admins will run their own one-off jobs in image registry namespace, but you never know. We should apply a common label to all Jobs created by the pruner CronJob and filter on that value.

Reply:
Since the cronjob creates the job, it looks like we are not able to set a label on it when it is created.  I will check if the cronjob sets a label on it already.

Reply:
I would assume that's something that can be set in the `CronJob` spec, in a similar fashion to labels in a `Deployment`

Reply:
updated in next push